libs/dao/src/main/java/com/akto/dao/testing/TestingRunResultDao.java

```diff
@@ -52,17 +52,20 @@ public List<TestingRunResult> fetchLatestTestingRunResult(Bson filters) {
     }
 
     public List<TestingRunResult> fetchLatestTestingRunResult(Bson filters, int limit) {
-        Bson projections = Projections.include(
-                                TestingRunResult.TEST_RUN_ID,
-                                TestingRunResult.API_INFO_KEY,
-                                TestingRunResult.TEST_SUPER_TYPE,
-                                TestingRunResult.TEST_SUB_TYPE,
-                                TestingRunResult.VULNERABLE,
-                                TestingRunResult.CONFIDENCE_PERCENTAGE,
-                                TestingRunResult.START_TIMESTAMP,
-                                TestingRunResult.END_TIMESTAMP,
-                                TestingRunResult.TEST_RUN_RESULT_SUMMARY_ID
-                        );
+        Bson projections = Projections.fields(
+                Projections.include(
+                        TestingRunResult.TEST_RUN_ID,
+                        TestingRunResult.API_INFO_KEY,
+                        TestingRunResult.TEST_SUPER_TYPE,
+                        TestingRunResult.TEST_SUB_TYPE,
+                        TestingRunResult.VULNERABLE,
+                        TestingRunResult.CONFIDENCE_PERCENTAGE,
+                        TestingRunResult.START_TIMESTAMP,
+                        TestingRunResult.END_TIMESTAMP,
+                        TestingRunResult.TEST_RUN_RESULT_SUMMARY_ID
+                ),
+                Projections.computed(TestingRunResult.ERRORS_LIST,Projections.computed("$arrayElemAt", Arrays.asList("$testResults.errors", 0)))
```

Comment:
check this query for cases where errors array is empty and when errors array does not exist.

Reply:
works fine